### PR TITLE
feat/Add toggle focus mode

### PR DESCRIPTION
**Pull request by rendi12345678**
https://github.com/user-attachments/assets/3ecbfc4d-caf0-421e-b582-7697f3318cc2

### CODE REVIEW

**Comment by MUYANGGUO** [comment]
Let us make this feature a button at the below next to the "word" , it will be only showing up when users at words mode, it also use the local storage to store the user preference. It will not show in the other mode.

**Comment by rendi12345678** [comment]
> Let us make this feature a button at the below next to the "word" , it will be only showing up when users at words mode, it also use the local storage to store the user preference. It will not show in the other mode.

ouhh okey sir, i would make it today

File: src/components/features/TypeBox/TypeBox.js [hunk 1]
```diff
@@ -1012,140 +1063,137 @@ const TypeBox = ({
...
-                      {char}
-                    </span>
-                  ))}
-                  {getExtraCharsDisplay(word, i)}
-                </span>
-              );
-            })}
-          </div>
+    <>
+      <SocialLinksModal status={status} />
```
Comment (MUYANGGUO) [inline src/components/features/TypeBox/TypeBox.js:1111]: looks like it shows up in the middle of the screen with a backdrop. can we make this less aggressive if it is going to be blocking the user view suddenly.

**Comment by rendi12345678** [comment]
https://github.com/user-attachments/assets/07fbaa45-63cc-4932-b1b8-4e220d389d24

done

File: src/components/features/TypeBox/TypeBox.js (hunk 1 above)
Comment (MUYANGGUO) [reply src/components/features/TypeBox/TypeBox.js:1111]: I think to ship the ultra zoom mode first, can temporarily remove this sociallinkmodal for now, we would probably ship this later in a separate pull request.
I think the logic is to 5 minutes users are seeing this modal shows up. And it is blocking users to see the status. 
So in that case, let's not make it to a module, but a regular div shows up everytime the game is finished. 
<Visual>
<Stats>
<Share(an icon group)>
when you click any of the social media button, then a modal pops up, to confirm the share. The V0 content would be simply "I have completed a typing challenge with XXX WPM", V1 would be a formatted svg / html/ image that we could create a certificate. 
So I think there is something we need to plan first before shipping in this pull request.

Comment (MUYANGGUO) [inline src/components/features/TypeBox/TypeBox.js:1111]:
```suggestion
```

File: src/components/common/FooterMenu.js [hunk 2]
```diff
@@ -168,6 +176,15 @@ const FooterMenu = ({
                   {WORD_MODE_LABEL}
                 </span>
               </IconButton>
+              {isWordGameMode && (
+                <IconButton onClick={toggleUltraZenMode}>
+                  <Tooltip title={ULTRA_ZEN_MODE}>
+                    <span className={getModeButtonClassName(isUltraZenMode)}>
+                      <ZoomInMapIcon fontSize="medium" />
```
Comment (MUYANGGUO) [inline src/components/common/FooterMenu.js:183]:
```suggestion
                      <ZoomInMapIcon fontSize="small" />
```

your call lol, let me know what do you think

File: .firebase/hosting.YnVpbGQ.cache [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+static/js/110.abadb285.chunk.js,1724337354184,5149fe8447d33c2b6c3c10bd7e0627d205e14e9b47ca558a0a81cf0847324716
+static/css/main.e6c13ad2.css.map,1724337354187,e845e6a28aaddc1902a601eeb9df71ac53bdd9c9a922c6d49febaf4035e42c47
+static/css/main.e6c13ad2.css,1724337354181,20ddccbcb290095bc023db15e46a257ec7d8ee4783d17704654f62d527804f71
+static/media/typeSoft.3e307070c0d43972460c.wav,1724337354181,1dade6a0dc93ff4a2f1658ae9d71850d6a217993e9f008bbdff6124c4f26c3c2
+static/media/cherryBlue.f4cad75e8a5792bb3d25.wav,1724337354181,22a637101f4b2aaef7335ed58a20e7db780ae522ef535506c05f1594b5e2b8d2
+static/media/keyboard.f9440324ce3fbfc79f5b.wav,1724337354181,1b248ff6cd5a2ffdc7f24c03f1dba874b01a1c285162a60589a6bb3e17f4b250
+static/media/WeChatSupport.d4d0b5b8d76008986829.png,1724337354181,a2dfd32d2be076a4e5c356ba97001789d403665f3dd90a01a631e9649f2fc32d
+static/js/845.d88f7a05.chunk.js.map,1724337354231,acc2fa06363004ca40b27d587ff87677dc97002c7d9a8c52db739433716f8b94
+static/js/main.97a30f1f.js,1724337354181,983f6120d7c952bedb2c692f8849dd9d364aba3a375cfee05ad3ee417e965411
+static/js/main.97a30f1f.js.map,1724337354227,48d92cc1410ce25a5de188dfe22b94546b79e18b47e81fa2be1583e7c96f5a62
```
Comment (MUYANGGUO) [inline .firebase/hosting.YnVpbGQ.cache:29]: are these the static build file cache for firebase hosting?

File: src/worker/trackHistoryWorker.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+// eslint-disable-next-line no-restricted-globals
+self.onmessage = function (e) {
```
Comment (MUYANGGUO) [inline src/worker/trackHistoryWorker.js:2]: awesome optimizations

File: src/components/features/TypeBox/TypeBox.js (hunk 1 above)
Comment (rendi12345678) [reply src/components/features/TypeBox/TypeBox.js]: yeah i made it hidden

**Review by MUYANGGUO — APPROVED** [review]

**Review by MUYANGGUO — COMMENTED** [review]
Approved, feel free to merge, and you can add the feature description in the readme.

**Review by MUYANGGUO — CHANGES_REQUESTED** [review]
let me block the merging first

**Review by MUYANGGUO — APPROVED** [review]

**Comment by rendi12345678** [comment]
great

**Pull request merged**